src/main/webapp/script.js

```diff
@@ -1,13 +1,333 @@
+  inp.addEventListener('input', function() {
+    let a; let b; let i; const val = this.value;
+
+    // Close any open autocomplete lists
+    closeAllLists();
+    if (!val) {
+      return false;
+    }
+
+    currentFocus = -1;
+
+    // Create a DIV element that will contain the items (values)
+    a = document.createElement('DIV');
+    a.setAttribute('id', this.id + 'autocomplete-list');
+    a.setAttribute('class', 'autocomplete-items');
+
+    // Append the DIV element as a child of the autocomplete container
+    this.parentNode.appendChild(a);
+
+    // Check if the item starts with the same letters as the text field value
+    for (i = 0; i < arr.length; i++) {
+      if (arr[i].substr(0, val.length).toUpperCase() == val.toUpperCase()) {
+        b = document.createElement('DIV');
+        b.innerHTML = '<strong>' + arr[i].substr(0, val.length) + '</strong>';
+        b.innerHTML += arr[i].substr(val.length);
+        b.innerHTML += '<input type=\'hidden\' value=\'' + arr[i] + '\'>';
+
+        // Execute a function when someone clicks on the item value (DIV element)
+        b.addEventListener('click', function(e) {
+          inp.value = this.getElementsByTagName('input')[0].value;
+          closeAllLists();
+        });
+
+        a.appendChild(b);
+      }
+    }
+  });
+
+  // Execute a function presses a key on the keyboard
+  inp.addEventListener('keydown', function(e) {
+    let x = document.getElementById(this.id + 'autocomplete-list');
+    if (x) x = x.getElementsByTagName('div');
+    if (e.keyCode == 40) {
+      // If the arrow DOWN key is pressed, increase currentFocus
+      currentFocus++;
+      addActive(x);
+    } else if (e.keyCode == 38) {  // up
+      // If the arrow UP key is pressed, decrease currentFocus
+      currentFocus--;
+      addActive(x);
+    } else if (e.keyCode == 13) {
+      // If the ENTER key is pressed, prevent form from being submitted
+      e.preventDefault();
+      if (currentFocus > -1) {
+        // Simulate click on active item
+        if (x) x[currentFocus].click();
+      }
+    }
+  });
+
+  // To classify an item as "active"
+  function addActive(x) {
+    if (!x) return false;
+
+    // Remove the "active" class on all items
+    removeActive(x);
+    if (currentFocus >= x.length) currentFocus = 0;
+    if (currentFocus < 0) currentFocus = (x.length - 1);
+
+    // Add class "autocomplete-active"
+    x[currentFocus].classList.add('autocomplete-active');
+  }
+
+  // To remove an item as "active" from all autocomplete items
+  function removeActive(x) {
+    for (let i = 0; i < x.length; i++) {
+      x[i].classList.remove('autocomplete-active');
+    }
+  }
+
+  // Close all autocomplete lists in the document
+  function closeAllLists(elmnt) {
+    const x = document.getElementsByClassName('autocomplete-items');
+    for (let i = 0; i < x.length; i++) {
+      if (elmnt != x[i] && elmnt != inp) {
+        x[i].parentNode.removeChild(x[i]);
+      }
+    }
+  }
```

Comment:
With an existing div purposed specifically for autocompletion this can be deleted and its use cases replaced with clearing the autocompletion div:
```javascript
let a = document.getElementById('autocomplete-list');
a.innerHTML = "";
```